ali1234/ugly
# Try to add support for TasBot

This should allow ugly to run on TasBot's new neopixel eyes.

To test this:

1. Make a venv and activate it
2. `pip install rpi-ws281x`
3. `git clone -b tasbot https://github.com/ali1234/ugly`
4. `cd ugly && pip install -e .`
5. `uglydemo -d tasbot -D legacy`

rpi_ws281x can use PWM, PCM, or SPI to drive LEDs. I think the method is selected automatically based on the GPIO you pick. See https://github.com/jgarff/rpi_ws281x for details.

This branch is currently set up for pin 18 and PWM DMA. You can change that to pin 10 to try SPI. The README suggests that it is faster.

Also note the code may be bit-rotted as I haven't worked on it in a long time. The dependencies it uses may be different now etc.

And also note, if you run `uglydemo -d tasbot -D terminal` you should see a preview 28x8 of what the LEDs should be doing, but on your terminal.
